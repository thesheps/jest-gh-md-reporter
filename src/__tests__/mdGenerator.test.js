const generator = require("../mdGenerator");
const {
	passingResults,
	failingResults,
	pendingResults,
} = require("./testData");

describe("MDGenerator", () => {
	const date = new Date("2022-01-01");

	it("Creates the expected snapshot for passing test", async () => {
		const report = await generator.generate(passingResults, date);
		expect(report).toMatchSnapshot();
	});

	it("Creates the expected snapshot for failing test", async () => {
		const report = await generator.generate(failingResults, date);
		expect(report).toMatchSnapshot();
	});

	it("Creates the expected snapshot for pending test", async () => {
		const report = await generator.generate(pendingResults, date);
		expect(report).toMatchSnapshot();
	});
});
