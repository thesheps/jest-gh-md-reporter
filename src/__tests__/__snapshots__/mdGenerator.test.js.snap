// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDGenerator Creates the expected snapshot for failing test 1`] = `
"# jest-gh-md-reporter Test Results

> **Started**: Sat, 01 Jan 2022 00:00:00 GMT

<center>

|Suites (2)|Tests (2)|
|:-:|:-:|
|![](https://img.shields.io/badge/Passed-1-green) | ![](https://img.shields.io/badge/Passed-1-green)|
|![](https://img.shields.io/badge/Failed-1-red) | ![](https://img.shields.io/badge/Failed-1-red)|
|![](https://img.shields.io/badge/Pending-1-orange) | ![](https://img.shields.io/badge/Pending-1-orange)|

---

<table>
<thead>
<tr>
<th>testFilePath</th>
<th></th>
<th></th>
<th>0.146</th>
</tr>
</thead>
<tbody>
<tr style=\\"background-color: pink; color: black\\">
<td><strong>Beans / On / Toast</strong></td>
<td><i>my lovely test</i></td>
<td>failed</td>
<td>0.146</td>
</tr>
</tbody>
</table>
</center>"
`;

exports[`MDGenerator Creates the expected snapshot for passing test 1`] = `
"# jest-gh-md-reporter Test Results

> **Started**: Sat, 01 Jan 2022 00:00:00 GMT

<center>

|Suites (2)|Tests (2)|
|:-:|:-:|
|![](https://img.shields.io/badge/Passed-1-green) | ![](https://img.shields.io/badge/Passed-1-green)|
|![](https://img.shields.io/badge/Failed-1-red) | ![](https://img.shields.io/badge/Failed-1-red)|
|![](https://img.shields.io/badge/Pending-1-orange) | ![](https://img.shields.io/badge/Pending-1-orange)|

---

<table>
<thead>
<tr>
<th>testFilePath</th>
<th></th>
<th></th>
<th>0.146</th>
</tr>
</thead>
<tbody>
<tr style=\\"background-color: lightgreen; color: black\\">
<td><strong>Beans / On / Toast</strong></td>
<td><i>my lovely test</i></td>
<td>passed</td>
<td>0.146</td>
</tr>
</tbody>
</table>
</center>"
`;

exports[`MDGenerator Creates the expected snapshot for pending test 1`] = `
"# jest-gh-md-reporter Test Results

> **Started**: Sat, 01 Jan 2022 00:00:00 GMT

<center>

|Suites (2)|Tests (2)|
|:-:|:-:|
|![](https://img.shields.io/badge/Passed-1-green) | ![](https://img.shields.io/badge/Passed-1-green)|
|![](https://img.shields.io/badge/Failed-1-red) | ![](https://img.shields.io/badge/Failed-1-red)|
|![](https://img.shields.io/badge/Pending-1-orange) | ![](https://img.shields.io/badge/Pending-1-orange)|

---

<table>
<thead>
<tr>
<th>testFilePath</th>
<th></th>
<th></th>
<th>0.146</th>
</tr>
</thead>
<tbody>
<tr style=\\"background-color: lightyellow; color: black\\">
<td><strong>Beans / On / Toast</strong></td>
<td><i>my lovely test</i></td>
<td>pending</td>
<td>0.146</td>
</tr>
</tbody>
</table>
</center>"
`;
